bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckablePullRequest.java

```diff
@@ -202,25 +204,30 @@ Hash amendManualReviewers(Hash commit, Namespace namespace, Hash original) throw
         }
     }
 
-    PullRequestCheckIssueVisitor createVisitor(Hash hash) throws IOException {
-        var checks = JCheck.checksFor(localRepo, hash);
+    PullRequestCheckIssueVisitor createVisitor(JCheckConfiguration conf, Hash hash) throws IOException {
+        var checks = JCheck.checksFor(conf, localRepo, hash);
         return new PullRequestCheckIssueVisitor(checks);
     }
 
-    void executeChecks(Hash localHash, CensusInstance censusInstance, PullRequestCheckIssueVisitor visitor,
-                       List<String> additionalConfiguration, Hash hash) throws IOException {
-        Optional<JCheckConfiguration> conf;
-        if (confOverride != null) {
-            conf = JCheck.parseConfiguration(confOverride, additionalConfiguration);
-        } else {
-            conf = JCheck.parseConfiguration(localRepo, hash, additionalConfiguration);
-        }
-        if (conf.isEmpty()) {
-            throw new RuntimeException("Failed to parse jcheck configuration at: " + hash + " with extra: " + additionalConfiguration);
+    Optional<JCheckConfiguration> parseJCheckConfiguration(Hash hash) throws IOException {
+        var original = confOverride == null ?
+            JCheck.parseConfiguration(localRepo, hash, List.of()) :
+            JCheck.parseConfiguration(confOverride, List.of());
+
+        if (original.isEmpty()) {
+            throw new IllegalStateException("Cannot parse JCheck configuration for commit with hash " + hash.hex());
         }
-        visitor.setConfiguration(conf.get());
-        try (var issues = JCheck.check(localRepo, censusInstance.census(), CommitMessageParsers.v1, localHash,
-                                       conf.get())) {
+
+        var botUser = pr.repository().forge().currentUser();
+        var additional = AdditionalConfiguration.get(original.get(), botUser, comments, reviewMerge);
+        return confOverride == null ?
+            JCheck.parseConfiguration(localRepo, hash, additional) :
+            JCheck.parseConfiguration(confOverride, additional);
+    }
+
+    void executeChecks(Hash localHash, CensusInstance censusInstance, PullRequestCheckIssueVisitor visitor, JCheckConfiguration conf) throws IOException {
```

Comment:
The parameter name `localHash` threw me off for a bit, maybe just `hash` in this context, unless I'm missing some context?

Reply:
Sure, renamed it t to `hash` in the latest version